src/migrator/utils.ts

```diff
@@ -152,3 +278,95 @@ export const migrationTablesExist = async (tx: Tx) => {
 	const result = await tx.tableList(where);
 	return result.rows.length === tables.length;
 };
+// Just update the "modified at" field to show liveness
+export const initMigrationStatus = async (
+	tx: Tx,
+	migrationStatus: MigrationStatus,
+): Promise<Result | undefined> => {
+	try {
+		const result = await tx.executeSql(
+			binds`
+INSERT INTO "migration status" ("migration key", "start time", "is backoff")
+VALUES (${1}, ${2}, ${3});
+`,
+			[
+				migrationStatus['migration_key'],
+				migrationStatus['start_time'],
+				migrationStatus['is_backoff'] ? 1 : 0,
+			],
+		);
+		return result;
+	} catch (err: any) {
+		// it already exists and we can ignore this error.
+	}
+};
+
+// Just update the "modified at" field to show liveness
+export const updateMigrationStatus = async (
+	tx: Tx,
+	migrationStatus: MigrationStatus,
+): Promise<Result | undefined> => {
+	try {
+		return await tx.executeSql(
+			binds`
+UPDATE "migration status"
+SET
+"run counter"	= ${1},
+"last run time" = ${2},
+"migrated rows" = ${3},
+"error counter" = ${4},
+"converged time" = ${5},
+"last error message" = ${6},
+"is backoff" = ${7}
+WHERE "migration status"."migration key" = ${8};`,
+			[
+				migrationStatus['run_counter'],
+				migrationStatus['last_run_time'],
+				migrationStatus['migrated_rows'],
+				migrationStatus['error_counter'],
+				migrationStatus['converged_time'],
+				migrationStatus['last_error_message'],
+				migrationStatus['is_backoff'] ? 1 : 0,
+				migrationStatus['migration_key'],
+			],
+		);
+	} catch (err) {
+		// we ignore all errors to not spam the api logs
+	}
+};
+
+export const readMigrationStatus = async (
+	tx: Tx,
+	migrationKey: string,
+): Promise<MigrationStatus | undefined> => {
+	try {
+		const { rows } = await tx.executeSql(
+			binds`
+SELECT *
+FROM "migration status"
+WHERE "migration status"."migration key" = ${1}
+LIMIT 1;`,
+			[migrationKey],
+		);
+
+		const data = rows[0];
+		if (data == null) {
+			return;
+		}
+
+		return {
+			migration_key: data['migration key'],
+			start_time: data['start time'],
+			last_run_time: data['last run time'],
+			run_counter: data['run counter'],
+			migrated_rows: data['migrated rows'],
+			error_counter: data['error counter'],
+			converged_time: data['converged time'],
+			last_error_message: data['last error message'],
+			is_backoff: data['is backoff'] === 1,
+		};
+	} catch (err) {
+		// we ignore all errors to not spam the api logs
```

Comment:
What sort of errors are you seeing?

Reply:
I was thinking about the possibility that the transaction is closed b/c the async migrator has faced any other issue. But as this is called only after the transactions has been created, I don't see any real issue. I followed the pattern here, that an async migrator should not fail the running instance but just silentlty fail.

Reply:
Yeah I agree it shouldn't fail the instance but it's the silencing that I'm questioning really - if something is failing in this section then it probably needs to be dealt with (at least I wouldn't expect any errors here as standard)

Reply:
Thus, a logging information would be fine, right?